Source/Util/PlayerRetriever.cs

```diff
@@ -0,0 +1,23 @@
+﻿using GameNetcodeStuff;
+
+namespace PlayerDogModel_Plus.Source.Util
+{
+    internal class PlayerRetriever
+    {
+        internal static PlayerControllerB GetPlayerFromClientId(ulong clientId)
+        {
+            if (StartOfRound.Instance.ClientPlayerList.TryGetValue(clientId, out int playerScriptIndex))
+            {
+                return StartOfRound.Instance.allPlayerScripts[playerScriptIndex];
+            }
+
+            foreach (PlayerControllerB player in StartOfRound.Instance.allPlayerScripts)
```

Comment:
Though this foreach should seldom be executed, it might be nice to add simple caching for it in the future.